test/test_xla_backend.py

```diff
@@ -0,0 +1,362 @@
+      pg = dist.new_group(ranks=ranks)
+    assert isinstance(pg, torch_xla.distributed.xla_backend.ProcessGroupXla)
+    assert pg.size() == len(ranks)
+    assert pg.rank() == ranks.index(world_rank)
+    assert pg._mesh == [[0, 4, 8], [1, 5, 9], [2, 6, 10], [3, 7, 11]]
+
+  def test_new_group_one_paticipant(self):
+    set_world_size(12)
+
+    ranks = [5]
+    world_rank = 5
+    set_world_rank(world_rank)
+    with new_group_barrier_disabled():
+      pg = dist.new_group(ranks=ranks)
+    assert isinstance(pg,
+                      torch_xla.distributed.xla_backend.ProcessGroupXla), str(
+                          type(pg))
+    assert pg.size() == 1
+    assert pg.rank() == 0
+    assert pg._mesh == [[0], [1], [2], [3], [4], [5], [6], [7], [8], [9], [10],
+                        [11]]
+
+  def test_new_group_entire_world(self):
+    set_world_size(12)
+
+    ranks = range(12)
+    world_rank = 5
+    set_world_rank(world_rank)
+    with new_group_barrier_disabled():
+      pg = dist.new_group(ranks=ranks)
+    assert isinstance(pg, torch_xla.distributed.xla_backend.ProcessGroupXla)
+    assert pg.size() == 12
+    assert pg.rank() == world_rank
+    assert pg._mesh == [[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]]
+
+  def test_new_group_invalid_horizontal(self):
+    set_world_size(12)
+
+    ranks = [4, 5, 6]
+    world_rank = 5
+    set_world_rank(world_rank)
+    with new_group_barrier_disabled():
+      with self.assertRaises(ValueError):
+        pg = dist.new_group(ranks=ranks)
+
+    ranks = [0, 1, 2, 3, 4]
+    world_rank = 2
+    set_world_rank(world_rank)
+    with new_group_barrier_disabled():
+      with self.assertRaises(ValueError):
+        pg = dist.new_group(ranks=ranks)
+
+    ranks = [7, 8, 9, 10]
+    world_rank = 9
+    set_world_rank(world_rank)
+    with new_group_barrier_disabled():
+      with self.assertRaises(ValueError):
+        pg = dist.new_group(ranks=ranks)
+
+  def test_new_group_invalid_vertical(self):
+    set_world_size(12)
+
+    ranks = [1, 5]
+    world_rank = 5
+    set_world_rank(world_rank)
+    with new_group_barrier_disabled():
+      with self.assertRaises(ValueError):
+        pg = dist.new_group(ranks=ranks)
+
+    ranks = [4, 7, 10]
+    world_rank = 4
+    set_world_rank(world_rank)
+    with new_group_barrier_disabled():
+      with self.assertRaises(ValueError):
+        pg = dist.new_group(ranks=ranks)
+
+  def test_new_group_invalid_ranks(self):
+    set_world_size(12)
+
+    # unevenly distributed
+    ranks = [1, 5, 10]
+    world_rank = 5
+    set_world_rank(world_rank)
+    with new_group_barrier_disabled():
+      with self.assertRaises(ValueError):
+        pg = dist.new_group(ranks=ranks)
+
+  def test_barrier(self):
+    # nothing to verify. Just run it through.
+    dist.barrier()
+
+  def test_unimplemented_ops(self):
+    unimplemented_ops = (
+        'reduce',
+        'allgather_coalesced',
+        'allreduce_coalesced',
+        'alltoall',
```

Comment:
xla and pt/xla has the `alltoall` implemantion. Any reason we don't use it here?

Reply:
This PR was aiming at enabling DeepSpeed. all_to_all is not needed there so we didn't prioritize it. We can add it later.